### DISCUSSION TITLE
what is the use of maxPages?

### DISCUSSION THREAD

**Discussion by kunalfleethawks** [sensedeep/dynamodb-onetable#300] [Q&A] [answered]
Hello @mobsense ,

What is the use of maxPages? I am not able to understand from the documentation? Can it be used with find() or just queryItems()? 

Thanks

**Comment by mobsense** [comment, marked as answer]
When doing a find with a limit and a filter condition, DynamoDB will scan ~1MB of data and may return nothing. You then use next/prev to get the next page. Find will do this for you and maxPages is the number of iterations the inner find() loop will do for you.

Without it, find could consume many I/Os while returning no data.

**Reply by mobsense** [reply]
I've updated the doc to clarify.
